### PR TITLE
fix: fix cluster incorrect keys status

### PR DESCRIPTION
**Pull request by BorysTheDev**
fix for #2989

### CODE REVIEW

File: src/server/cluster/incoming_slot_migration.h [hunk 1]
```diff
@@ -47,13 +47,16 @@ class IncomingSlotMigration {
     return source_id_;
   }
 
+  size_t GetKeysNumber() const;
```
Comment (chakaz) [inline src/server/cluster/incoming_slot_migration.h:50]: How about:
```suggestion
  size_t GetKeyCount() const;
```
?

File: src/server/cluster/cluster_family.cc [hunk 2]
```diff
@@ -660,26 +660,6 @@ static string_view StateToStr(MigrationState state) {
   return "UNDEFINED_STATE"sv;
 }
 
-static uint64_t GetKeyCount(const SlotRanges& slots) {
```
Comment (adiholden) [inline src/server/cluster/cluster_family.cc:663]: where is the bug? where did we count all keys and not only the migrated keys?

Comment (BorysTheDev) [reply src/server/cluster/cluster_family.cc:663]: when migration is finished and we start to process new requests the number of keys will be changed and we will show incorrect statistic

File: src/server/cluster/incoming_slot_migration.h (hunk 1 above)
Comment (BorysTheDev) [reply src/server/cluster/incoming_slot_migration.h:50]: It doesn't matter for me, if I do update of the PR I will apply this

File: src/server/cluster/incoming_slot_migration.cc [hunk 3]
```diff
@@ -120,8 +121,10 @@ IncomingSlotMigration::~IncomingSlotMigration() {
 }
 
 void IncomingSlotMigration::Join() {
+  // TODO add timeout
   bc_->Wait();
-  state_ = MigrationState::C_FINISHED;
+  state_.store(MigrationState::C_FINISHED);
+  keys_number_ = GetKeyCount(slots_);
```
Comment (adiholden) [inline src/server/cluster/incoming_slot_migration.cc:127]: Is this optimization so we wont calc slots key count again or should we must do this calc here to show the correct key count?
if this is an optimization I think its not so important as we the GetKeyCount is kind of the same of dbsize calc right? just gets stats from each shard

File: src/server/cluster/cluster_family.cc (hunk 2 above)
Comment (adiholden) [reply src/server/cluster/cluster_family.cc:663]: Oh I see

File: src/server/cluster/incoming_slot_migration.cc (hunk 3 above)
Comment (adiholden) [reply src/server/cluster/incoming_slot_migration.cc:127]: Ok got the answer to this question from your comment above

Comment (adiholden) [reply src/server/cluster/incoming_slot_migration.cc:127]: Ok got the answer to this question from your comment above

File: src/server/cluster/incoming_slot_migration.h [hunk 4]
```diff
@@ -47,13 +47,16 @@ class IncomingSlotMigration {
...
+
  private:
   std::string source_id_;
   Service& service_;
   std::vector<std::unique_ptr<ClusterShardMigration>> shard_flows_;
   SlotRanges slots_;
   std::atomic<MigrationState> state_ = MigrationState::C_NO_STATE;
   Context cntx_;
+  size_t keys_number_ = 0;
```
Comment (adiholden) [inline src/server/cluster/incoming_slot_migration.h:59]: please add a comment here and in the outgoing class about this counter

**Review by adiholden — APPROVED** [review]

**Pull request merged**